[dashboard] Fix overlay box size

CaseyHillers opened the pull request:
* On web, it's inheriting the mobile version, which present the wrong size
* Use static sizes instead of inheriting
* Generalize the task overlay tests to make them easier to change when the TaskBox size changes

Fixes https://github.com/flutter/flutter/issues/137705

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

tvolkert commented:
This doesn't seem the right fix.  What was the failure mode?  Was it that it wasn't finding a `TaskBox` in the hierarchy and thus falling back to the default cell size?

CaseyHillers commented:
> This doesn't seem the right fix. What was the failure mode? Was it that it wasn't finding a TaskBox in the hierarchy and thus falling back to the default cell size?

Fixed! Moved the TaskBox to the top of the tree to ensure it's in the context when the OverlayEntry is created

tvolkert reviewed (approved):
LGTM!
